### PR TITLE
test: migrate part of form tests

### PR DESCRIPTION
**Pull request by li-jia-nan**
[[中文版模板 / Chinese template](https://github.com/ant-design/ant-design/blob/master/.github/PULL_REQUEST_TEMPLATE/pr_cn.md)]

### 🤔 This is a ...

- [ ] New feature
- [ ] Bug fix
- [ ] Site / documentation update
- [ ] Demo update
- [ ] Component style update
- [ ] TypeScript definition update
- [ ] Bundle size optimization
- [ ] Performance optimization
- [ ] Enhancement feature
- [ ] Internationalization
- [ ] Refactoring
- [ ] Code style optimization
- [x] Test Case
- [ ] Branch merge
- [ ] Other (about what?)

### 🔗 Related issue link

- close #37118 

### 📝 Changelog
test: migrate part of form tests

| Language   | Changelog |
| ---------- | --------- |
| 🇺🇸 English | test: migrate part of form tests |
| 🇨🇳 Chinese | test 迁移 |

### ☑️ Self-Check before Merge

⚠️ Please check all items below before review. ⚠️

- [x] Doc is updated/provided or not needed
- [x] Demo is updated/provided or not needed
- [x] TypeScript definition is updated/provided or not needed
- [x] Changelog is provided or not needed

### CODE REVIEW

File: components/form/__tests__/index.test.tsx [hunk 1]
```diff
@@ -478,37 +468,42 @@ describe('Form', () => {
...
+    await userEvent.clear(screen.getByLabelText('bamboo'));
 
-    wrapper.find('input[type="checkbox"]').simulate('change', { target: { checked: true } });
-    wrapper.update();
-    expect(wrapper.find('.ant-form-item-required')).toHaveLength(1);
+    // should show alert says that the field is required
+    await expect(screen.findByRole('alert')).resolves.toHaveTextContent("'bamboo' is required");
   });
 
-  describe('should show related className when customize help', () => {
```
Comment (zombieJ) [inline components/form/__tests__/index.test.tsx:488]: 这个 describe 怎么没了？

Comment (li-jia-nan) [reply components/form/__tests__/index.test.tsx:488]: 补上了

File: components/form/__tests__/index.test.tsx [hunk 2]
```diff
@@ -593,61 +616,62 @@ describe('Form', () => {
...
     );
 
     expect(shouldNotRender).toHaveBeenCalledTimes(1);
     expect(shouldRender).toHaveBeenCalledTimes(1);
 
-    formRef.current!.setFieldsValue({ light: 'bamboo' });
-    await Promise.resolve();
+    formRef.current?.setFieldsValue({ light: 'bamboo' });
+
+    await waitFakeTimer();
```
Comment (zheeeng) [inline components/form/__tests__/index.test.tsx:651]: 这里是不是导致 test 失败了？

Comment (li-jia-nan) [reply components/form/__tests__/index.test.tsx:651]: 这里有什么问题嘛？我本地是跑过了的

Comment (zheeeng) [reply components/form/__tests__/index.test.tsx:651]: https://github.com/ant-design/ant-design/pull/37930

这里 test 失败了，shouldRender 应该 call 两次，测试只 call 了一次

**Pull request merged**